### DISCUSSION TITLE
Train ner component on subset of labels in training corpus

### DISCUSSION THREAD

**Discussion by kinghuang** [explosion/spaCy#9480] [Help: Coding & Implementations] [answered]
I have a corpus that is fully labelled with the target entities. But, some of these are meant for [entity_ruler](https://spacy.io/api/entityruler) components to label, while others are for an [ner](https://spacy.io/api/entityrecognizer) component to learn and predict. Is there a way to configure a `ner` component to only learn specific labels? Or, do I have to produce a filtered version of the corpus for it?

I see that the `ner` component can take a `labels` argument during initialization, and [spacy init labels](https://spacy.io/api/cli#init-labels) can generate the content for this. But, it doesn't look like that's an easy way for me to include or exclude certain labels.

**Comment by polm** [comment, marked as answer]
No, we don't have a feature for this. I think the code to create a filtered DocBin should be minimal.

**Reply by kinghuang** [reply]
Ok, I'll filter the data that goes into making the DocBin files in my project. Thanks!
